src/shared/parameter_v2/c/config_test.cpp

```diff
@@ -0,0 +1,264 @@
+#include <gtest/gtest.h>
+
+#include <boost/filesystem.hpp>
+#include <regex>
+
+#include "clang-c/Index.h"
+
+
+extern "C"
+{
+#include "shared/parameter_v2/c/config.h"
+#include "shared/parameter_v2/c/parameter.h"
+}
+
+#include <cstring>
+
+#include "yaml-cpp/yaml.h"
+
+/**
+ * This function converts a config file names into config struct names.
+ *
+ * For example, a config file name maybe "example_config" gets converted
+ * to "ExampleConfig"
+ *
+ * @param The snake_case string to convert to CamelCase
+ * @return Converted string to CamelCase
+ */
+std::string toCamelCase(const std::string& snake_case_input)
+{
+    std::string ret_str = "";
+    ret_str += (char)toupper(snake_case_input[0]);
+
+    for (unsigned i = 1; i < snake_case_input.length() - 1; i++)
+    {
+        if (snake_case_input[i] == '_')
+        {
+            ret_str += (char)toupper(snake_case_input[i + 1]);
+            i++;
+        }
+        else
+        {
+            ret_str += snake_case_input[i];
+        }
+    }
+
+    ret_str += snake_case_input[snake_case_input.length() - 1];
```

Comment:
this line can be removed and the loop condition can be changed to i < snake_case_input.length()

Reply:
I initially had that incase someone put a _ at the end of the input

However, thats not valid snake case, so I added a pre-condition to make sure that its valid

Fixed